metricbeat/module/kibana/cluster_actions/cluster_actions.go

```diff
@@ -54,38 +54,48 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 
-	actionsHTTP, err := helper.NewHTTP(ms.BaseMetricSet)
-	if err != nil {
-		return nil, err
-	}
-	kibanaVersion, err := kibana.GetVersion(actionsHTTP, kibana.ClusterActionsPath)
-	if err != nil {
-		return nil, err
-	}
-
-	isMetricsAPIAvailable := kibana.IsActionsAPIAvailable(kibanaVersion)
-	if !isMetricsAPIAvailable {
-		const errorMsg = "the %v cluster actions is only supported with Kibana >= %v. You are currently running Kibana %v"
-		return nil, fmt.Errorf(errorMsg, ms.FullyQualifiedName(), kibana.ActionsAPIAvailableVersion, kibanaVersion)
-	}
-
 	return &MetricSet{
-		MetricSet:   ms,
-		actionsHTTP: actionsHTTP,
+		MetricSet: ms,
 	}, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right format
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
 func (m *MetricSet) Fetch(r mb.ReporterV2) (err error) {
+	if err = m.init(); err != nil {
+		return err
+	}
+
 	if err = m.fetchMetrics(r); err != nil {
 		return fmt.Errorf("error trying to get cluster action data from Kibana: %w", err)
 	}
 
 	return nil
 }
 
+func (m *MetricSet) init() error {
+	actionsHTTP, err := helper.NewHTTP(m.BaseMetricSet)
+	if err != nil {
+		return err
+	}
```

Comment:
Does this function make a network call or is the potential failure only related to misconfiguration ? If the latter it would be good to move it back to the constructor. It will also allow to set it up only once